src/rpc/misc.cpp

```diff
@@ -323,11 +323,13 @@ UniValue createmultisig(const UniValue& params, bool fHelp)
 
 UniValue createwitnessaddress(const UniValue& params, bool fHelp)
 {
-    if (fHelp || params.size() < 1 || params.size() > 1)
+    if (fHelp || params.size() < 1 || params.size() > 1 || Params().NetworkIDString() == "main")
```

Comment:
Please don't do a string test on the chain params. If you need a specific condition, add a ChainParams::AllowCreateWitness method or so.